src/plugins/discover/public/application/main/components/log_explorer/log_explorer_main_route.tsx

```diff
@@ -0,0 +1,240 @@
+import {
+  AnalyticsNoDataPageKibanaProvider,
+  AnalyticsNoDataPage,
+} from '@kbn/shared-ux-page-analytics-no-data';
+import {
+  SavedSearch,
+  getSavedSearch,
+  getSavedSearchFullPathUrl,
+} from '../../../../services/saved_searches';
+import { getState } from '../../services/discover_state';
+import { loadIndexPattern, resolveIndexPattern } from '../../utils/resolve_index_pattern';
+import { LogExplorerMainApp } from './log_explorer_main_app';
+import { getRootBreadcrumbs, getSavedSearchBreadcrumbs } from '../../../../utils/breadcrumbs';
+import { LoadingIndicator } from '../../../../components/common/loading_indicator';
+import { DiscoverError } from '../../../../components/common/error_alert';
+import { useDiscoverServices } from '../../../../hooks/use_discover_services';
+import { getUrlTracker } from '../../../../kibana_services';
+
+const LogExplorerMainAppMemoized = memo(LogExplorerMainApp);
+
+interface DiscoverLandingParams {
+  id: string;
+}
+
+interface Props {
+  isDev: boolean;
+}
+
+export function DiscoverLogExplorerRoute(props: Props) {
+  const history = useHistory();
+  const services = useDiscoverServices();
+  const { isDev } = props;
+  const {
+    core,
+    chrome,
+    uiSettings: config,
+    data,
+    toastNotifications,
+    http: { basePath },
+    dataViewEditor,
+  } = services;
+  const [error, setError] = useState<Error>();
+  const [savedSearch, setSavedSearch] = useState<SavedSearch>();
+  const indexPattern = savedSearch?.searchSource?.getField('index');
+  const [indexPatternList, setIndexPatternList] = useState<Array<SavedObject<DataViewAttributes>>>(
+    []
+  );
+  const [hasESData, setHasESData] = useState(false);
+  const [hasUserDataView, setHasUserDataView] = useState(false);
+  const [showNoDataPage, setShowNoDataPage] = useState<boolean>(false);
+  const { id } = useParams<DiscoverLandingParams>();
+
+  useExecutionContext(core.executionContext, {
+    type: 'application',
+    page: 'app',
+    id: id || 'new',
+  });
+
+  const loadDefaultOrCurrentIndexPattern = useCallback(
+    async (searchSource: ISearchSource) => {
+      try {
+        const hasUserDataViewValue = await data.dataViews.hasData
+          .hasUserDataView()
+          .catch(() => false);
+        const hasESDataValue =
+          isDev || (await data.dataViews.hasData.hasESData().catch(() => false));
+        setHasUserDataView(hasUserDataViewValue);
+        setHasESData(hasESDataValue);
+
+        if (!hasUserDataViewValue) {
+          setShowNoDataPage(true);
+          return;
+        }
+
+        const defaultDataView = await data.dataViews.getDefaultDataView();
+
+        if (!defaultDataView) {
+          setShowNoDataPage(true);
+          return;
+        }
+
+        const { appStateContainer } = getState({ history, uiSettings: config });
+        const { index } = appStateContainer.getState();
+        const ip = await loadIndexPattern(index || '', data.dataViews, config);
```

Comment:
`loadIndexPattern` doesn't return an index pattern directly, it returns `IndexPatternData` which contains the list, but also some extra properties like `stateVal` and `stateValFound`, which are used in other parts of code to determine certain things (like showing a toast).